src/main/app/src/components/Purchasing.js

```diff
@@ -41,25 +54,163 @@ const orderRows = [
   { id: 4, vendorName: 'Vendor1', items: '????', timestamp: "01/31/2021", cost: "$100" }
 ];
 
-const LoadedView = () => {
+function TextMaskCustom(props) {
+  const { inputRef, ...other } = props;
+
+  return (
+    <MaskedInput
+      {...other}
+      ref={(ref) => {
+        inputRef(ref ? ref.inputElement : null);
+      }}
+      mask={['(', /[1-9]/, /\d/, /\d/, ')', ' ', /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, /\d/]}
+      placeholderChar={'\u2000'}
+      showMask
+    />
+  );
+}
+
+TextMaskCustom.propTypes = {
+  inputRef: PropTypes.func.isRequired,
+};
+
+const PhoneNumberField = () => {
+  const [values, setValues] = React.useState({
+    textmask: '(  )    -    ',
+    numberformat: '1320',
+  });
+
+  const handleChange = (event) => {
+    setValues({
+      ...values,
+      [event.target.name]: event.target.value,
+    });
+  };
+  return (
+    <FormControl>
+        <InputLabel>Phone Number</InputLabel>
+        <Input
+          value={values.textmask}
+          onChange={handleChange}
+          name="textmask"
+          inputComponent={TextMaskCustom}
+        />
+      </FormControl>
+  );
+};
+
+const ProvinceDrawer = () =>{
+  const [province, setProvince] = React.useState('');
+
+  const handleChange = (event) => {
+    setProvince(event.target.value);
+  };
+  return(
+    <div  style={{ width: "250px" }}>
+      <FormControl>
+        <Select
+          value={province}
+          onChange={handleChange}
+        >
+          <MenuItem value={"QC"}>QC</MenuItem>
+          <MenuItem value={"ON"}>ON</MenuItem>SS
+        </Select>
+      </FormControl>
+    </div>
+  );
+};
+
+function LoadedView() {
+  const [open, setOpen] = React.useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const AddCustomerDialogBox = () => {
+    return (
+      <Dialog open={open} onClose={handleClose} fullWidth maxWidth='sm'>
```

Comment:
The dialog looks pretty good, but the form is a little rough:
![Screenshot_20210209_090531](https://user-images.githubusercontent.com/52555183/107374870-155de800-6ab6-11eb-8183-8210c894d0a9.png)

In general:
- The fields in the form are too close together. Could you add some bigger `margin` to each field or something?
- Fields need some validation - the form should not be submit-able with invalid postal code format, phone number, etc.
- I think we don't need the "Please Do Not Leave Any Fields Empty!" warning at the top. The field names should have an asterix next to the required fields and if the user tries to submit, empty required fields should be highlighted in some way to indicate that those fields must be filled, like so:
![image](https://user-images.githubusercontent.com/52555183/107376847-5a831980-6ab8-11eb-8b2b-3c12d14e1997.png)
Or like:
![image](https://user-images.githubusercontent.com/52555183/107376917-6d95e980-6ab8-11eb-98e0-0e7db54ad643.png)

Reply:
There is a "required" attribute for text fields , will look into that and see how it works

Reply:
Added required label to name field , but is not enforced